### PR TITLE
Prow upgrade mechanism

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Adding script for upgrading all prow deployments using in place rollout and upgrading images
- Rollback mechanism in case of upgrade errors
- [WIP] ProwJob for periodic upgrades (as we are not sure yet, what should be the tigger event)

**Related issue(s)**
* #148

### CODE REVIEW

File: prow/scripts/upgrade-prow.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+set -o errexit
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck disable=SC1090
+source "${SCRIPT_DIR}/library.sh"
+
+authenticate
+
+TAG=$(curl https://raw.githubusercontent.com/kubernetes/test-infra/master/prow/cluster/starter.yaml | grep image | grep deck | awk '{print $2}' | cut -d ':' -f2)
```
Comment (aszecowka) [inline prow/scripts/upgrade-prow.sh]: What will happen if they add a new required parameter to an application? This script does not help us in such a situation.

Comment (aszecowka) [reply prow/scripts/upgrade-prow.sh]: Have you checked how other projects approach that topic? Have you asked on a slack channel what are their recommendations?

Comment (Demonsthere) [reply prow/scripts/upgrade-prow.sh]: Yes, I have look into istio and knative, you can find my comment in #148 
Sadly, I don't think there is a good way of automating the deployment manifests. Istio and knaive both do it semi manually and update only the image tag.

**Pull request closed without merging**